File: Practice/Recursion/AlgoExpert/NthFibonaci/solution3.cpp
using namespace std;

int getNthFib(int n) {
  // Write your code here.
	int lastTwo[2] = {0, 1};
	int counter = 3;
	
	while(counter <=n){
		int nextFib = lastTwo[0]+ lastTwo[1];
		lastTwo[0]=lastTwo[1];
		lastTwo[1]=nextFib;
		counter +=1;
	}
	if(n>1){
		return lastTwo[1];
	}else{
		return lastTwo[0];
	}
}

File: Basic/AlgoExpert/StairCaseTransversal/solution3.py
def staircaseTraversal(height, maxSteps):
    # Write your code here.
    waysToTop = [0 for _ in range(height+1)]
	waysToTop[0] = 1
	waysToTop[1] = 1
	
	for currentHeight in range(2, height + 1):
		step =1
		while step <= maxSteps and step <= currentHeight:
			waysToTop[currentHeight] = waysToTop[currentHeight] + waysToTop[currentHeight - step]
			step +=1
	return waysToTop[height]

# O(n* k) time | O(n) space  - where n is height k is max steps